lib/dotcom_web/components/trip_planner/itinerary_group.ex

```diff
@@ -60,13 +58,15 @@ defmodule DotcomWeb.Components.TripPlanner.ItineraryGroup do
           Similar trips depart at <%= Enum.map(@summary.next_starts, &format_datetime_short/1)
           |> Enum.join(", ") %>
         </div>
-        <button class="btn-link font-semibold underline" phx-click={JS.toggle(to: "##{@group_id}")}>
+        <button
+          class="btn-link font-semibold underline"
+          phx-click="show_itinerary_details"
```

Comment:
In react-land, this would be a callback that would be explicitly passed into this component - `onDetailsClick` or something like that. I feel like we need _some_ analogous strategy for that here, because I really don't like having `"show_itinerary_details"` being hard-coded in both this file and `trip_planner.ex`.

Do you have any suggestions? I can also experiment before merging, maybe by passing in a `phx-details-click` `assign` or something like that?

Reply:
ha, I was going to link to this exact doc! can totally do that.

Reply:
I didn't end up passing in a callback because `itinerary_group` isn't a live component (pretty sure only live components can do anything meaningful with callbacks?). I did pass `details_click_event` and `target` in as assigns, which sort of accomplishes the same goal.